### DISCUSSION TITLE
How to create an aptos wallet from mnemonic (seed phrases) instead of private key?

### DISCUSSION THREAD

**Discussion by 0x-j** [aptos-labs/aptos-developer-discussions#12] [Questions] [answered]
As title

**Comment by 0x-j** [comment, marked as answer]
Checkout the [wallet standard doc](https://aptos.dev/standards/wallets/#mnemonics-phrases) using ts-sdk, basically you can get private key from mnemonic then get wallet from privatekey. 

```typescript
  static fromDerivePath(path: string, mnemonics: string): AptosAccount {
    if (!AptosAccount.isValidPath(path)) {
      throw new Error("Invalid derivation path");
    }

    const normalizeMnemonics = mnemonics
      .trim()
      .split(/\s+/)
      .map((part) => part.toLowerCase())
      .join(" ");

    const { key } = derivePath(path, bytesToHex(bip39.mnemonicToSeedSync(normalizeMnemonics)));

    return new AptosAccount(new Uint8Array(key));
  }
```

If you want to use python, here's an [open source repo](https://github.com/satisfywithmylife/aptos_wallet_mnemonic/tree/main) made by community.

**Reply by slyofzero** [reply]
what is path here, and I can't get the AptosAccount class from `@aptos-labs/ts-sdk`

**Reply by 0xmaayan** [reply]
Can refer to this unit test and see what arguments the function expects and how to define them. 

For example, given this wallet https://github.com/aptos-labs/aptos-ts-sdk/blob/main/tests/unit/helper.ts#L9 can derive the account as follow https://github.com/aptos-labs/aptos-ts-sdk/blob/main/tests/unit/account.test.ts#L133-L142.

Note: the function mentioned here has changed to `Account.fromDerivationPath` - can find it here https://github.com/aptos-labs/aptos-ts-sdk/blob/main/src/core/account/Account.ts#L202
